### ISSUE TITLE
Add support for the RISC-V CPU architecture

### ISSUE THREAD

**Issue by ghuntley** [coder/coder#4708]
Let's add support for running our `coder` CLI on the RISC-V CPU architecture (https://riscv.org/). 

It should be relatively straight forward (`GOARCH=riscv GOOS=linux`) but there's a chance it could be quite involved (topics of stubbing if 3rd party libraries are not supported) thus the importance to communicate early if it looks like this isn't going to be straight forward. 

0. build `coder/coder` for RISC-V
1. update our GitHub actions pipelines to publish this RISC-V binary when other binaries for other CPU architectures are published.
2. test your binary on RISC-V hardware.

**Comment by teezzan** [comment]
I'd love to take a shot at this.

**Comment by ghuntley** [comment]
Make it so!

**Comment by teezzan** [comment]
Not at all. I thought to emulate it on my device one way or the other. Do you have any idea or suggestions?

**Comment by ghuntley** [comment]
> Not at all. I thought to emulate it on my device one way or the other. Do you have any idea or suggestions?

Unfortunately you'll need access to RISCV hardware to do this task.

**Comment by bpmct** [comment]
Gonna close this for now - however I think it would be really cool if Coder eventually had RISK V support. Please re-open / leave a comment if you get to this

**Comment by jeehoonkang** [comment]
Hi! I set up a local coder instance and tried to use my RISC-V machine as a docker workspace provider, and it failed due to this issue :) I just wonder if anyone else is interested in this issue as well.

I have RISC-V machines and I can provide them for test. Though I think we can test with qemu-virtualized RISC-V machines on x86 physical machines.

**Comment by bpmct** [comment]
I did some tinkering, both with cross compilations and directly on RISC-V hardware (Sipeed LicheePi 4A) and was able to get the slim binary compiled with few hurdles. I'm working on building a riscv64 container image that can do these builds. 

Building the full binary introduces some complications. Some dependency stuff which I'm sure we can resolve (looks like we need to compile dprint for Risc V), but more importantly Terraform and OpenTofu (and the providers) do not support Risc-V either.

**Comment by liberodark** [comment]
Hi,

If you need to have access to risc-v hardware.
I can give you access and those for free as part of the development of coder.
I would be happy to see your application add riscv64 support.
You can request a server here : https://open-riscv-initiative.com/

Best regards

**Comment by matifali** [comment]
Hi @liberodark I have submitted a request. Thanks for the generous offer. We can definitely make use of that to build and test compatibility with riscv64 hardware.

**Comment by liberodark** [comment]
Hi im send your access.

**Comment by matifali** [comment]
@liberodark Could you check your email? I have an access issue. Thanks.

**Comment by liberodark** [comment]
Hi @matifali explain me your issue ?

Best Regards

**Comment by matifali** [comment]
I was able to access the server over ssh but don't have sudo access nor do I know the password. So I couldn't install anything. 
Username is the same as my GitHub username.

**Comment by bpmct** [comment]
I was able to cross-compile the slim binary for RISC-V. Wouldn't really recommend running the fat binary on RISC-V at this point since Terraform and the underlying providers don't have great RISC-V support. I'm sure it is _possible_ but I am running the control plane on x86 with the ability to spin up RISC-V workspaces with Docker.

I'm hosting slim binary on: https://files.ben.coder.app/coder-slim_linux_riscv64 

Branch: https://github.com/coder/coder/compare/main...add-riscv-support

Custom template code (just using Docker)
```hcl
provider docker {
  # coderd will `ssh` into the RISC-V box
  host     = "ssh://sipeed@lpi4a:22"
}

resource "docker_container" "workspace" {
  # ..  # Use the docker gateway if the access URL is 127.0.0.1
entrypoint = [
  "sh",
  "-c",
  <<EOF
curl -fsSL --compressed https://files.ben.coder.app/coder-slim_linux_riscv64 -o coder
chmod +x coder
./coder agent
EOF
]
  env        = [
    "CODER_AGENT_URL=${data.coder_workspace.me.access_url}",
    "CODER_AGENT_AUTH=token",
    "CODER_AGENT_TOKEN=${coder_agent.main.token}",
  ]
```

Minimal container image:

```Dockerfile
FROM ubuntu

ENV DEBIAN_FRONTEND=noninteractive

# Install dependencies (including libatomic1 and busybox for tar extraction)
RUN apt-get update && apt-get install -y curl busybox
```
